sycl/test-e2e/Basic/launch_queries/max_num_work_groups.cpp

```diff
@@ -0,0 +1,223 @@
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out
+
+#include <sycl/detail/core.hpp>
+
+#include <cassert>
+#include <cstdint>
+
+namespace syclex = sycl::ext::oneapi::experimental;
+using namespace sycl::info::device;
+using namespace sycl::info::kernel_device_specific;
+
+using value_type = int64_t;
+
+namespace kernels {
+
+template <class T, size_t Dim>
+using sycl_global_accessor =
+    sycl::accessor<T, Dim, sycl::access::mode::read_write,
+                   sycl::access::target::global_buffer>;
+
+class TestKernel {
+public:
+  static constexpr bool HasLocalMemory{false};
+
+  TestKernel(sycl_global_accessor<value_type, 1> acc) : acc_{acc} {}
+
+  void operator()(sycl::nd_item<1> item) const {
+    const auto gtid = item.get_global_linear_id();
+    acc_[gtid] = gtid + 42;
+  }
+
+private:
+  sycl_global_accessor<value_type, 1> acc_;
+};
+
+class TestLocalMemoryKernel {
+public:
+  static constexpr bool HasLocalMemory{true};
+
+  TestLocalMemoryKernel(sycl_global_accessor<value_type, 1> acc,
+                        sycl::local_accessor<value_type, 1> loc_acc)
+      : acc_{acc}, loc_acc_{loc_acc} {}
+
+  void operator()(sycl::nd_item<1> item) const {
+    const auto ltid = item.get_local_linear_id();
+    const auto gtid = item.get_global_linear_id();
+    if (ltid < loc_acc_.size()) {
+      loc_acc_[ltid] = ltid + 42;
+      item.barrier(sycl::access::fence_space::local_space);
+      acc_[gtid] = loc_acc_[ltid];
+    } else {
+      acc_[gtid] = 0;
+    }
+  }
+
+private:
+  sycl_global_accessor<value_type, 1> acc_;
+  sycl::local_accessor<value_type, 1> loc_acc_;
+};
+
+} // namespace kernels
+
+namespace {
+
+template <class KernelName>
+int test_max_num_work_groups(sycl::queue &q, const sycl::device &dev) {
+  const auto ctx = q.get_context();
+  auto bundle = sycl::get_kernel_bundle<sycl::bundle_state::executable>(ctx);
+  auto kernel = bundle.template get_kernel<KernelName>();
+
+  const size_t maxWorkGroupSize =
+      kernel.template get_info<work_group_size>(dev);
+  const size_t NumWorkItems = maxWorkGroupSize * maxWorkGroupSize;
+
+  size_t workGroupSize = 32;
+  size_t localMemorySizeInBytes{0};
+  if constexpr (KernelName::HasLocalMemory) {
+    localMemorySizeInBytes = workGroupSize * sizeof(value_type);
+  }
+
+  sycl::buffer<value_type, 1> buf{sycl::range<1>{NumWorkItems}};
+
+  // Tests
+
+  // ==================== //
+  // Test 1 - return type //
+  // ==================== //
+  std::cout << "\nTest 1" << std::endl;
```

Comment:
I would prefer no output from tests in positive scenario, because strictly speaking, `lit` still has to process it, so why waste machine resources?

Reply:
Me too. That logging was left unintentionally. Will remove.